### ISSUE TITLE
Overlay isn't set on sj.run

### ISSUE THREAD

**Issue by backdround** [woosaaahh/sj.nvim#28]
**Versions used for**
 - Neovim - v0.9.4
 - SJ.nvim - 7ca6186

**Describe the bug**
Overlay isn't set on `sj.run`

**To Reproduce**
- open neovim with file
- wait for lsp to start (~1/2 sec)
- perform `sj.run`
- exit `sj.run`
- move screen-view several lines down (don't know how many)
- perform `sj.run`
- see that overlay isn't set

**Expected behavior**
Overlay is set anytime `sj.run` is performed

**Gif**
![js-issue](https://github.com/woosaaahh/sj.nvim/assets/17349169/eff469dd-a752-4d7c-9ab4-c0fe691fd0c9)

**Additional context**
- The bug appears only with any lsp plugins enabled (nvim-lspconfig / null-ls.nvim)
- in lua files (it doesn't appear in golang files)
- SjOverlay and other highlights are set

By the way, thanks for the plugin. I'm really looking forward to using it

**Comment by woosaaahh** [comment]
Hi @backdround !

Thanks for reporting this issue and taking the time to detail it, it helped a lot to debug !

I was able to reproduce this behavior with `v0.9.4` but not `v0.10.0` (nightly).
Apparently, this is an issue with Neovim which was fixed by this commit : https://github.com/neovim/neovim/commit/b04286a187d57c50f01cd36cd4668b7a69026579.
The fix is not available in Neovim `v0.9.4`.


Based on your details, I found that the issue happens when you have LSP semantic tokens enabled.
I disable them so I never experienced this problem.
I don't use `lspconfig` or any LSP plugins but here is how I disable LSP semantic tokens (at the end of the code below) : 

```lua
function on_attach(client, bufnr)
	local opts = { buffer = bufnr }

	keymap("n", "[d", vim.diagnostic.goto_prev, opts)
	keymap("n", "]d", vim.diagnostic.goto_next, opts)
	keymap("n", "<localleader>ge", vim.diagnostic.open_float, opts)
	keymap("n", "<localleader>gq", vim.diagnostic.setqflist, opts)

	keymap("n", "<localleader>ga", vim.lsp.buf.code_action, opts)
	keymap("n", "<localleader>gd", vim.lsp.buf.definition, opts)
	keymap("n", "<localleader>gh", vim.lsp.buf.hover, opts)
	keymap("n", "<localleader>gr", vim.lsp.buf.references, opts)

	keymap("n", "<localleader>gs", vim.lsp.buf.signature_help, opts)
	keymap("i", "<A-,>", vim.lsp.buf.signature_help, opts)

	--- Don't let LSP semantic tokens take over Tree-sitter highlightings
	client.server_capabilities.semanticTokensProvider = false  -- <-------- The important line
end
```

Maybe you can switch to Neovim `nightly` or keep using Neovim `v0.9.4` and disable LSP semantic tokens ?
I'm not sure I'll try (or be able) to fix this issue in `sj.nvim`.

**Comment by backdround** [comment]
Thank you @woosaaahh !

I've just tried it on nightly Neovim `v0.10.0` and it works as expected!
